packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,306 @@
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+            html<TableColumnMapping>`
+                ${repeat(x => x.mappedTemplate,
+                    html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                )}
+            `
+        )}
+    `
+)}
+```
+
+For each mapping, if the key matches the value for this row, get the mapping's contained `template` and render its `innerHTML`.
+
+#### `nimble-mapping-string/number/boolean`
+
+```HTML
+<template slot="mapping">
+    <slot
+        ${slotted({ property: 'mappedTemplate', filter: x => x instanceof HTMLTemplateElement })}
+    >
+    </slot>
+</template>
+```
+
+### Grouping
+
+Will support grouping by the record value. The grouping header will display the rendered html mapped to that value. If multiple different values map to the same rendered html (e.g. icon), there may be separate groups that visually look like they should be combined into one group. We could instead render the record value into the header, but that value is not otherwise exposed to the user. It may not even have much meaning (e.g. some numeric code). I think it is more reasonable to display the rendered html.
+
+### Sorting
+
+Sorting has the same issue as grouping, where visually identical cells could end up not sorted contiguously. Again, I believe this is reasonable.
+
+### Sizing
+
+We will support fixed or fractional widths for this column type.
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component. _Initial attempts at prototyping resulted in exceptions being thrown and no icons being rendered. This is an open issue._
+
+### Visual Appearance
+
+Appearance can vary based on mapped html, but can only include text, icons, and Nimble spinner.
+
+---
+
+## Implementation
+
+_Important aspects of the planned implementation with careful consideration of web standards and integration._
+
+_Highlight any alternative implementations you considered in each section._
+
+_If you think a section doesn't apply or don't know what to write, please DO NOT delete it. Either mark it "N/A" or leave it blank and the Nimble team can help you fill it in._
+
+### States
+
+N/A
+
+### Accessibility
+
+Text, icons, and spinner are not interactive and cannot receive keyboard focus.
+
+The cell will not introduce any items into the accessibility tree other than the elements in the mapped template. Those items will already have appropriate accessibility tree representations.
+
+We will instruct clients to provide an accessible name for each icon/spinner, either via the `title` or `aria-label` attribute. `aria-labelledby` is not an option, because it requires defining an `id` which will not be unique when duplicated accross multiple rows.
```

Comment:
Setting `title` does not create a label in the accessibility tree. Are there alternatives that would make this easier for clients? For example, could the icon mapping element have an attribute that was the label, and nimble would set that as the `aria-label` and as the `title`?

Reply:
> Setting title does not create a label in the accessibility tree.

I believe it does (though, as a "fallback"). As [this W3C doc](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/) states, "The title attribute may be used as the element's fallback accessible name."

> could the icon mapping element have an attribute that was the label...

I initially had this, but the question becomes which element(s) to set that on. We could set it on a `div` wrapping the whole thing, but that doesn't actually provide an accessible name to the wrapped elements. If we're directing the client to set `description` on our mapping element, I think we might as well just direct them to set the `title` or `aria-label` on the appropriate element instead.

Reply:
> I believe it does (though, as a "fallback"). As this W3C doc states, "The title attribute may be used as the element's fallback accessible name."

I don't have an explicit alternative proposal yet. I'll just point out that the same cited doc says not to rely on title as a fallback:

> Rule 4: Avoid Browser Fallback
> When authors do not specify an accessible name using an element or attribute that is intended for naming, browsers attempt to help assistive technology users by resorting to fallback methods for generating a name. For example, the HTML title and placeholder attributes are used as last resort sources of content for accessible names. Because the purpose of these attributes is not naming, their content typically yields low quality accessible names that are not effective.


We had a similar discussion in the Web Chapter meeting on the [Hijacking Screenreaders with CSS presentation](https://www.youtube.com/watch?v=kk-03n3Y4WI) (good watch, we have hit many of the issues it talks about) where in regards to `text-transform: uppercase`:

> Maybe we should strive for is minimizing the heuristics we rely on. I bet the browsers will each consistently report it as the text ADD in the accessibility tree (we can verify that in chrome and firefox devtools, I assume Safari has an accessibility tree inspector too?), but we know that's not what we want to be in the tree. So even if heuristics work pretty well today for certain acronyms in certain languages, we probably don't want to rely screen readers correcting the wrong thing being shown in the accessibility tree.

Reply:
Good point. In that case, I think we would document that the user should set `aria-label` and optionally `title` as well on the element(s) in their template.

Reply:
Now we will be setting `aria-label` ourselves, based on the provided `label` value.